packages/webview_flutter/ios/Classes/FlutterWebView.m

```diff
@@ -113,6 +134,16 @@ - (void)updateJsMode:(NSNumber*)mode {
   }
 }
 
+- (void)updateUserAgent:(NSString*)userAgent {
+  if (@available(iOS 9.0, *)) {
+    [_webView setCustomUserAgent:userAgent];
+  } else if (userAgent) {
+    [[NSUserDefaults standardUserDefaults] registerDefaults:@{@"UserAgent" : userAgent}];
+    WKWebViewConfiguration* configuration = _webView.configuration;
+    _webView = [[WKWebView alloc] initWithFrame:_webView.frame configuration:configuration];
```

Comment:
We can't do this as it won't change the webview reference that the platform views controller is holding.
Even if we could update the engine's reference, doing this will lose any state the webview is currently in.

I think a better approach for iOS < 9 would be to do the user defaults trick prior to the  construction of the webview in `FLTWebViewController`'s `initWithFrame` (ohh I just noticed we have a typo there).

And just not support changing the userAgent for iOS < 9 (as WkWebView doesn't support it).

Note that we should make sure to restore the UserAgent value after constructing the webview.